pkg/compact/compact.go

```diff
@@ -885,7 +898,9 @@ func NewBucketCompactor(
 // Compact runs compaction over bucket.
 func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 	defer func() {
-		if IsHaltError(rerr) {
+		// Leave the compact directory for inspection if it is a halt error
```

Comment:
so we don't check halt. I think you mean any error, which is ok - as long it's not full disk error ;p

Reply:
also everywhere else